rosidl_typesupport_connext_cpp/src/dds_utilities.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rosidl_typesupport_connext_cpp/dds_utilities.hpp>
+
+bool
+get_datareader_qos(DDSDomainParticipant * participant, DDS_DataReaderQos & datareader_qos)
```

Comment:
It may be true that it should be in rmw_connext_cpp instead though. The question is where is it used from. If it is used from the message specific code, then it would need to be here or as you suggested moved to rmw and used there, passing the result to the message specific code.